spec/solver_spec.rb

```
@@ -0,0 +1,12 @@
+require_relative '../testing'
+
+describe 'Solver' do
+  context '#factorial' do
+   it 'Takes one argument, an integer, and returns its factorial' do
+    expect(Solver.factorial(5)).to eql 120
+   end
+   it 'Test if the argument passed is an integer if not throw' do 
+    expect{ Solver.factorial("hi") }.to raise_error(NoMethodError)
+   end
+  end
```

- [x] It will be great to test if the number passed as an argument is a negative value. When this is the case then it should raise an error.

---

Thank you so much for your contribution I will make sure to add it!